include/ensmallen_bits/nsga2/nsga2.hpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file nsga2.hpp
+ * @author Sayan Goswami
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_NSGA2_NSGA2_HPP
+#define ENSMALLEN_NSGA2_NSGA2_HPP
+
+namespace ens {
+
+class NSGA2 {
+ public:
+  NSGA2(const size_t populationSize = 100,
+        const size_t maxGenerations = 2000,
+        const double crossoverProb = 0.6,
+        const double mutationProb = 0.3,
+        const double mutationStrength = 1e-3,
+        const double epsilon = 1e-6);
+
+  template<typename MultiobjectiveFunctionType,
+           typename MatType,
+           typename... CallbackTypes>
+  std::vector<MatType> Optimize(MultiobjectiveFunctionType& objectives,
+                   MatType& iterate,
+                   CallbackTypes&&... callbacks);
+
+  size_t PopulationSize() const { return populationSize; }
+  size_t& PopulationSize() { return populationSize; }
+
+  size_t MaxGenerationSize() const { return maxGenerations; }
+  size_t& MaxGenerationSize() { return maxGenerations; }
+
+  double CrossoverProb() const { return crossoverProb; }
+  double& CrossoverProb() { return crossoverProb; }
+
+  double MutationProb() const { return mutationProb; }
```

Comment:
I think for now we should name this `MutationProbability()`, since that's what we are using in `CNE`. Also perhaps we should change `CrossoverProb()` to `CrossoverRate()` since that's what we are using in `DE`. Just to make our API consistent :)